lib/features/auth/repository/auth_repository.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter_appauth/flutter_appauth.dart';
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:gruene_app/app/constants/config.dart';
+import 'package:jwt_decoder/jwt_decoder.dart';
+import 'package:logger/logger.dart';
+
+class AuthRepository {
+  final FlutterAppAuth _appAuth = FlutterAppAuth();
+  final FlutterSecureStorage _secureStorage = FlutterSecureStorage();
+  final Logger _logger = Logger();
+
+  Future<bool> signIn() async {
+    try {
+      final AuthorizationTokenResponse result = await _appAuth.authorizeAndExchangeCode(
+        AuthorizationTokenRequest(
+          Config.clientId,
+          'com.example.grueneapp://oauthredirect',
+          issuer: Config.issuer,
+          scopes: ['openid', 'profile', 'email'],
+        ),
+      );
+
+      await _secureStorage.write(key: 'access_token', value: result.accessToken);
+      await _secureStorage.write(key: 'id_token', value: result.idToken);
+      await _secureStorage.write(key: 'refresh_token', value: result.refreshToken);
```

Comment:
:upside_down_face: Perhaps it would make sense to create constants for the keys? Very minor though.
```suggestion
      const accessTokenKey = 'access_token'
      const idTokenKey = 'id_token'
      const refresh_token = 'refresh_token'
      await _secureStorage.write(key: 'access_token', value: result.accessToken);
      await _secureStorage.write(key: 'id_token', value: result.idToken);
      await _secureStorage.write(key: 'refresh_token', value: result.refreshToken);
```

Reply:
Thank you for this idea, I created constants for the keys.